src/models/CallOptions.ts

```diff
@@ -0,0 +1,93 @@
+import { CallType } from './CallType';
+import { CustomMetaData } from './CustomMetaData';
+import { DTMFInput } from './DTMFInput';
+
+class CallOptions {
+  receiverCuid: string;
+  callContext: string;
+  customMetaData: CustomMetaData;
+
+  constructor(
+    receiverCuid: string,
+    callContext: string,
+    customMetaData: CustomMetaData
+  ) {
+    this.receiverCuid = receiverCuid;
+    this.callContext = callContext;
+    this.customMetaData = customMetaData;
+  }
+
+  static fromDict(dict: any) {
+    const callerCuid = dict.receiverCuid;
+    const callContext = dict.callContext;
+    const customMetaData = CustomMetaData.fromDict(dict.customMetaData);
+    const callOptions = new CallOptions(
+      callerCuid,
+      callContext,
+      customMetaData
+    );
+
+    return callOptions;
+  }
+}
+
+class M2PCallOptions extends CallOptions {
+  campaignId: string;
+  campaignEndTime: string;
+  campaignLabelList: string[];
```

Comment:
mark `campaignLabelList` and `dtmfInputList` as undefined  as well

Reply:
Done